### ISSUE TITLE
L006/TSQL: rule wants to split certain compound comparison operators

### ISSUE THREAD

**Issue by jpers36** [sqlfluff/sqlfluff#2209]
This is caused by #1965.

## Expected Behaviour

L006 does not flag/fix the following code:

```
select *
from test
where 1 != 1
```

## Observed Behaviour

L006 changes the code to
```
select *
from test
where 1 ! = 1
```

## Dialect

tsql

## Version

latest main

**Comment by jpers36** [comment]
@jpy-git would #2197 fix this?

**Comment by jpy-git** [comment]
@jpers36 yeah that'll be fixed by that PR. Should have some time to finish it off later this week 👍
